fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -560,6 +569,9 @@ private void setDateValues(Parameter p, DateTime dateTime) {
      * @param end
      */
     private void setDateValues(Parameter p, java.time.Instant start, java.time.Instant end) {
+        if (start == null) {
+            return;
+        }
```

Comment:
what if start is not null, but end is?

also, maybe update the javadoc to say what happens for a null start or end?

Reply:
maybe it should be more like this?
```
        if (start != null) {
            Timestamp startTime = Timestamp.from(start);
            p.setValueDateStart(startTime);
            p.setValueDate(startTime);
            p.setTimeType(TimeType.UNKNOWN);
        }

        if (end != null) {
            Timestamp implicitEndExclusive = Timestamp.from(end);
            // TODO: Is it possible to avoid this by using <= or BETWEEN instead of < when constructing the query?
            Timestamp implicitEndInclusive = convertToInclusiveEnd(implicitEndExclusive);
            p.setValueDateEnd(implicitEndInclusive);
        }
```

        Timestamp implicitEndExclusive = Timestamp.from(end);
        // TODO: Is it possible to avoid this by using <= or BETWEEN instead of < when constructing the query?
        Timestamp implicitEndInclusive = convertToInclusiveEnd(implicitEndExclusive);
        p.setValueDateEnd(implicitEndInclusive);
```